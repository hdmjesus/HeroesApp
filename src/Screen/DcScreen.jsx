import React from "react";
import { HeroesList } from "../Components/heroes/List";

export const DcScreen = () => {
  return (
    <div className="container mt-5">
      <h1 className="m-3">Dc Heroes</h1>
      <hr />
      <HeroesList publisher={"DC Comics"} />
    </div>
  );
};
