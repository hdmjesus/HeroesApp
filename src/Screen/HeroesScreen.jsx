import React from "react";
import { Redirect, useParams } from "react-router";
import { getHeroById } from "../selectors/getHeroesById";

export const HeroesScreen = ({ history }) => {
  const { heroid } = useParams();
  const hero = getHeroById(heroid);
  console.log(hero);
  if (!hero) {
    return <Redirect to="/" />;
  }
  const { id, superhero, publisher, alter_ego, first_appearance, characters } =
    hero;
  const handleReturn = () => {
    if (history.length <= 2) {
      publisher === "Marvel Comics" && history.push("/");
      publisher === "DC Comics" && history.push("/dc");
    } else {
      history.goBack();
    }
  };
  return (
    <div className="animate__animated animate__fadeInLeft p-7 md:flex">
      <div className="">
        <img className=" " src={`../images/${heroid}.webp`} alt={heroid} />
      </div>
      <div className="container  flex mt-4 flex-col items-center md:items-start   ">
        <h3 className="md:ml-28 text-lg uppercase font-bold">{superhero}</h3>
        <ul className="list-group list-group-flush text-xs md:text-base md:m-auto md:mb-5 md:mt-5">
          <li className="list-group-item">
            {" "}
            <b>Alter Ego: </b> {alter_ego}
          </li>
          <li className="list-group-item">
            {" "}
            <b>Publisher: </b> {publisher}
          </li>
          <li className="list-group-item">
            {" "}
            <b>First Apearance: </b> {first_appearance}
          </li>
          <li className="list-group-item">
            {" "}
            <b>Characters: </b> {characters}
          </li>
        </ul>

        <button
          className="btn btn-outline-info mt-3 m-auto"
          onClick={handleReturn}
        >
          Return
        </button>
      </div>
    </div>
  );
};
