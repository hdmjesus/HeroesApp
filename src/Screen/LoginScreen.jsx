import React from "react";

export const LoginScreen = ({ history }) => {
  const handleLogin = () => {
    history.replace("/");
  };
  return (
    <div className="container mt-5">
      <h1 className="m-3">LoginScreen</h1>
      <hr />
      <button className="btn btn-primary m-4" onClick={handleLogin}>
        Login
      </button>
    </div>
  );
};
