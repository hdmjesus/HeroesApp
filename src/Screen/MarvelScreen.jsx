import React from "react";
import { HeroesList } from "../Components/heroes/List";

export const MarvelScreen = () => {
  return (
    <div className="container mt-5">
      <h1 className="m-3">Marvel Comics</h1>
      <hr />

      <HeroesList publisher={"Marvel Comics"} />
    </div>
  );
};
