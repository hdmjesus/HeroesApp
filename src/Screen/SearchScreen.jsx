import React, { useMemo, useState } from "react";
import queryString from "query-string";
import { useLocation } from "react-router";
import { HeroeCard } from "../Components/heroes/HeroeCard";
import { useForm } from "../Hooks/useForm";
import { getHeroesByName } from "../selectors/getHeroesByName";
export const SearchScreen = ({ history }) => {
  const [error, setError] = useState(false);
  const location = useLocation();
  const { q = " " } = queryString.parse(location.search);

  const [formValue, handleInputChange, reset] = useForm({ searchText: q });
  const { searchText } = formValue;

  const heroesFiltered = useMemo(() => getHeroesByName(searchText), [q]);

  const handleError = () => {
    if (q === "" || q === " ") {
      setError(false);
      console.log("error false");
    }
    if (heroesFiltered.length === 0 && q !== "") {
      setError(true);
      console.log(" error true");
    }
  };
  const handleSearch = (e) => {
    e.preventDefault();
    handleError();
    history.push(`?q=${searchText}`);
  };

  return (
    <div>
      <div className=" flex gap-5 flex-col md:flex-row container  mt-4">
        <div className="md:w-5/12">
          <h4 className="m-3">Search your heroe</h4>
          <hr />

          <form
            className="container mt-3 flex flex-col"
            action=""
            onSubmit={handleSearch}
          >
            <input
              className="form-control outline-none "
              type="text"
              name="searchText"
              id=""
              placeholder="Find you hero"
              onChange={handleInputChange}
              autoComplete="off"
              value={searchText}
            />
            <button
              type="submit"
              className="btn w-2/4 m-auto  md:w-full mt-3 btn-block btn-outline-primary"
            >
              Search
            </button>
          </form>
        </div>

        <div className="md:w-8/12 mt-4 ">
          {" "}
          <h4 className="mv-3">Results</h4>
          <hr />
          {q === " " && error == false ? (
            <div className="alert alert-info w-full m-2"> Search a heroe</div>
          ) : null}
          {heroesFiltered.length === 0 && !error && (
            <div className="alert alert-danger m-2">
              <p>
                Error, personaje <b>{q}</b> no econtrado
              </p>
            </div>
          )}
          <div className="flex flex-wrap m-auto">
            {heroesFiltered.map((hero) => (
              <HeroeCard key={hero.id} {...hero} />
            ))}
          </div>
        </div>
      </div>
    </div>
  );
};
