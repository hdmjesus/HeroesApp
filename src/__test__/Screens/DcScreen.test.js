import { shallow, mount } from "enzyme";
import React from "react";
import { MemoryRouter } from "react-router";
import { DcScreen } from "../../Screen/DcScreen";

describe("Pruebas en <DcScreen/>", () => {
  test("debe mostrarse correctamente el componente", () => {
    const wrapper = mount(
      <MemoryRouter>
        <DcScreen />
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("h1").text().trim()).toBe("Dc Heroes");
  });
});
