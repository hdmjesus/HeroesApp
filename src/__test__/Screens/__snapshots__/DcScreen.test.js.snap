// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <DcScreen/> debe mostrarse correctamente el componente 1`] = `
<div
  className="container mt-5"
>
  <h1
    className="m-3"
  >
    Dc Heroes
  </h1>
  <hr />
  <div
    className="flex justify-center flex-wrap animate__animated animate__fadeIn"
  >
    <div
      className="card mt-5 mb-3 m-1  text-xs"
      style={
        Object {
          "maxWidth": 350,
          "minWidth": 150,
          "padding": 0,
        }
      }
    >
      <div
        className="row g-0"
      >
        <div
          className="col-md-4"
        >
          <img
            alt="superhero"
            className="card-img"
            src="./images/dc-batman.webp"
          />
        </div>
        <div
          className="col-md-8"
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Batman
            </h5>
            <p
              className="card-text"
            >
              Bruce Wayne
            </p>
            <p
              className="card-text"
            >
               
              <small
                className="text-muted"
              >
                 
                Detective Comics #27
              </small>
            </p>
            <a
              className="text-blue-500"
              href="/hero/dc-batman"
              onClick={[Function]}
            >
              Mas...
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card mt-5 mb-3 m-1  text-xs"
      style={
        Object {
          "maxWidth": 350,
          "minWidth": 150,
          "padding": 0,
        }
      }
    >
      <div
        className="row g-0"
      >
        <div
          className="col-md-4"
        >
          <img
            alt="superhero"
            className="card-img"
            src="./images/dc-superman.webp"
          />
        </div>
        <div
          className="col-md-8"
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Superman
            </h5>
            <p
              className="card-text"
            >
              Kal-El
            </p>
            <p
              className="card-text"
            >
               
              <small
                className="text-muted"
              >
                 
                Action Comics #1
              </small>
            </p>
            <a
              className="text-blue-500"
              href="/hero/dc-superman"
              onClick={[Function]}
            >
              Mas...
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card mt-5 mb-3 m-1  text-xs"
      style={
        Object {
          "maxWidth": 350,
          "minWidth": 150,
          "padding": 0,
        }
      }
    >
      <div
        className="row g-0"
      >
        <div
          className="col-md-4"
        >
          <img
            alt="superhero"
            className="card-img"
            src="./images/dc-flash.webp"
          />
        </div>
        <div
          className="col-md-8"
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Flash
            </h5>
            <p
              className="card-text"
            >
              Jay Garrick
            </p>
            <p
              className=""
            >
               
              Jay Garrick, Barry Allen, Wally West, Bart Allen
            </p>
            <p
              className="card-text"
            >
               
              <small
                className="text-muted"
              >
                 
                Flash Comics #1
              </small>
            </p>
            <a
              className="text-blue-500"
              href="/hero/dc-flash"
              onClick={[Function]}
            >
              Mas...
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card mt-5 mb-3 m-1  text-xs"
      style={
        Object {
          "maxWidth": 350,
          "minWidth": 150,
          "padding": 0,
        }
      }
    >
      <div
        className="row g-0"
      >
        <div
          className="col-md-4"
        >
          <img
            alt="superhero"
            className="card-img"
            src="./images/dc-green.webp"
          />
        </div>
        <div
          className="col-md-8"
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Green Lantern
            </h5>
            <p
              className="card-text"
            >
              Alan Scott
            </p>
            <p
              className=""
            >
               
              Alan Scott, Hal Jordan, Guy Gardner, John Stewart, Kyle Raynor, Jade, Sinestro, Simon Baz
            </p>
            <p
              className="card-text"
            >
               
              <small
                className="text-muted"
              >
                 
                All-American Comics #16
              </small>
            </p>
            <a
              className="text-blue-500"
              href="/hero/dc-green"
              onClick={[Function]}
            >
              Mas...
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card mt-5 mb-3 m-1  text-xs"
      style={
        Object {
          "maxWidth": 350,
          "minWidth": 150,
          "padding": 0,
        }
      }
    >
      <div
        className="row g-0"
      >
        <div
          className="col-md-4"
        >
          <img
            alt="superhero"
            className="card-img"
            src="./images/dc-arrow.webp"
          />
        </div>
        <div
          className="col-md-8"
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Green Arrow
            </h5>
            <p
              className="card-text"
            >
              Oliver Queen
            </p>
            <p
              className="card-text"
            >
               
              <small
                className="text-muted"
              >
                 
                More Fun Comics #73
              </small>
            </p>
            <a
              className="text-blue-500"
              href="/hero/dc-arrow"
              onClick={[Function]}
            >
              Mas...
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card mt-5 mb-3 m-1  text-xs"
      style={
        Object {
          "maxWidth": 350,
          "minWidth": 150,
          "padding": 0,
        }
      }
    >
      <div
        className="row g-0"
      >
        <div
          className="col-md-4"
        >
          <img
            alt="superhero"
            className="card-img"
            src="./images/dc-wonder.webp"
          />
        </div>
        <div
          className="col-md-8"
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Wonder Woman
            </h5>
            <p
              className="card-text"
            >
              Princess Diana
            </p>
            <p
              className="card-text"
            >
               
              <small
                className="text-muted"
              >
                 
                All Star Comics #8
              </small>
            </p>
            <a
              className="text-blue-500"
              href="/hero/dc-wonder"
              onClick={[Function]}
            >
              Mas...
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card mt-5 mb-3 m-1  text-xs"
      style={
        Object {
          "maxWidth": 350,
          "minWidth": 150,
          "padding": 0,
        }
      }
    >
      <div
        className="row g-0"
      >
        <div
          className="col-md-4"
        >
          <img
            alt="superhero"
            className="card-img"
            src="./images/dc-martian.webp"
          />
        </div>
        <div
          className="col-md-8"
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Martian Manhunter
            </h5>
            <p
              className="card-text"
            >
              J'onn J'onzz
            </p>
            <p
              className=""
            >
               
              Martian Manhunter
            </p>
            <p
              className="card-text"
            >
               
              <small
                className="text-muted"
              >
                 
                Detective Comics #225
              </small>
            </p>
            <a
              className="text-blue-500"
              href="/hero/dc-martian"
              onClick={[Function]}
            >
              Mas...
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card mt-5 mb-3 m-1  text-xs"
      style={
        Object {
          "maxWidth": 350,
          "minWidth": 150,
          "padding": 0,
        }
      }
    >
      <div
        className="row g-0"
      >
        <div
          className="col-md-4"
        >
          <img
            alt="superhero"
            className="card-img"
            src="./images/dc-robin.webp"
          />
        </div>
        <div
          className="col-md-8"
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Robin/Nightwing
            </h5>
            <p
              className="card-text"
            >
              Dick Grayson
            </p>
            <p
              className="card-text"
            >
               
              <small
                className="text-muted"
              >
                 
                Detective Comics #38
              </small>
            </p>
            <a
              className="text-blue-500"
              href="/hero/dc-robin"
              onClick={[Function]}
            >
              Mas...
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card mt-5 mb-3 m-1  text-xs"
      style={
        Object {
          "maxWidth": 350,
          "minWidth": 150,
          "padding": 0,
        }
      }
    >
      <div
        className="row g-0"
      >
        <div
          className="col-md-4"
        >
          <img
            alt="superhero"
            className="card-img"
            src="./images/dc-blue.webp"
          />
        </div>
        <div
          className="col-md-8"
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Blue Beetle
            </h5>
            <p
              className="card-text"
            >
              Dan Garret
            </p>
            <p
              className=""
            >
               
              Dan Garret, Ted Kord, Jaime Reyes
            </p>
            <p
              className="card-text"
            >
               
              <small
                className="text-muted"
              >
                 
                Mystery Men Comics #1
              </small>
            </p>
            <a
              className="text-blue-500"
              href="/hero/dc-blue"
              onClick={[Function]}
            >
              Mas...
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card mt-5 mb-3 m-1  text-xs"
      style={
        Object {
          "maxWidth": 350,
          "minWidth": 150,
          "padding": 0,
        }
      }
    >
      <div
        className="row g-0"
      >
        <div
          className="col-md-4"
        >
          <img
            alt="superhero"
            className="card-img"
            src="./images/dc-black.webp"
          />
        </div>
        <div
          className="col-md-8"
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Black Canary
            </h5>
            <p
              className="card-text"
            >
              Dinah Drake
            </p>
            <p
              className=""
            >
               
              Dinah Drake, Dinah Lance
            </p>
            <p
              className="card-text"
            >
               
              <small
                className="text-muted"
              >
                 
                Flash Comics #86
              </small>
            </p>
            <a
              className="text-blue-500"
              href="/hero/dc-black"
              onClick={[Function]}
            >
              Mas...
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
